avocado/core/task/runtime.py

```diff
@@ -46,3 +55,132 @@ def __eq__(self, other):
         if isinstance(other, RuntimeTask):
             return hash(self) == hash(other)
         return False
+
+
+class RuntimeGraph:
+
+    def __init__(self, test_suite, job_id, status_server_uri):
+        self.test_suite = test_suite
+        self.job_id = job_id
+        self.status_server_uri = status_server_uri
+
+    def _get_requirements_runtime_tasks(self, runnable, prefix):
+        if runnable.requirements is None:
+            return
+
+        # creates the runnables for the requirements
+        requirements_runnables = RequirementsResolver.resolve(runnable)
+        requirements_runtime_tasks = []
+        # creates the tasks and runtime tasks for the requirements
+        for requirement_runnable in requirements_runnables:
+            name = '%s-%s' % (requirement_runnable.kind,
+                              requirement_runnable.kwargs.get('name'))
+            # the human UI works with TestID objects, so we need to
+            # use it to name other tasks
+            task_id = TestID(prefix,
```

Comment:
Not related to this PR (and not your fault), but this signals that it's time to revise the `TestID` concept, probably making it a more generic term and implementation.

Maybe a `TestID` has really got to be a `TaskID`, which can happen to be a test (and thus will have its PASS/FAIL result counted, result files kept in `test-results`, etc).

I'll open an issue about it.